sdks/python/apache_beam/io/iobase.py

```diff
@@ -1093,3 +1095,212 @@ def process(self, context):
     if self.counter >= self.count:
       self.counter -= self.count
     yield self.counter, context.element
+
+
+class ConcatSource(BoundedSource):
+  """A ``BoundedSource`` that can group a set of ``BoundedSources``."""
+
+  class ConcatRangeTracker(RangeTracker):
```

Comment:
This is not a general-purpose range tracker that others should be using, so it makes more sense to keep it as an inner class of ConcatSource.